src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3421,6 +3444,15 @@ public override BoundNode VisitConversion(BoundConversion node)
 
             (BoundExpression operand, Conversion conversion) = RemoveConversion(node, includeExplicitConversions: true);
             TypeSymbolWithAnnotations operandType = VisitRvalueWithResult(operand);
+
+            switch (node.ConversionKind)
+            {
+                case ConversionKind.ImplicitNullable:
+                case ConversionKind.ExplicitNullable:
+                    TrackNullableStateIfNullableConversion(node);
```

Comment:
>TrackNullableStateIfNullableConversion(node); [](start = 20, length = 45)

Could this have an undesired effect on the ApplyConversion call? It changes the state before we "processed" the conversion. Consider if it would be more appropriate to move this after the ApplyConversion call.